plugins/in_syslog/syslog_prot.c

```diff
@@ -32,6 +34,80 @@ static inline void consume_bytes(char *buf, int bytes, int length)
     memmove(buf, buf + bytes, length - bytes);
 }
 
+#define DYNAMIC_TAG_MAX 128
+
+// find and fill the value by msgpack format, e.g. \xA5ident\xA7my-appX...
+// https://github.com/msgpack/msgpack/blob/master/spec.md#str-format-family
+static int fill_field_value(char const *data, size_t const data_size, char const *name, char *out_val, char const *out_limit) {
```

Comment:
writing raw msgpack bytes is strongly discouraged, please use normal API for the purpose.

please re-write this function with normal unpackers and conditionals.

Reply:
@edsiper , fixed